library/ssl_cli.c

```diff
@@ -3090,8 +3090,8 @@ static int ssl_parse_server_key_exchange( mbedtls_ssl_context *ssl )
             mbedtls_ssl_send_alert_message(
                 ssl,
                 MBEDTLS_SSL_ALERT_LEVEL_FATAL,
-                MBEDTLS_SSL_ALERT_MSG_ILLEGAL_PARAMETER );
-            return( MBEDTLS_ERR_SSL_BAD_HS_SERVER_KEY_EXCHANGE );
+                MBEDTLS_SSL_ALERT_MSG_INTERNAL_ERROR );
+            return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
```

Comment:
Yes, it's another case where we don't have sufficient information to pick the right error code. 

I chose `INTERNAL_ERROR` because its documentation matches well, namely "e.g. an unexpected failure in a low-level module".

In line with the other occasions, perhaps we should go with `HANDSHAKE_FAILURE` unless we know better.